### ISSUE TITLE
brew install <formula> yields `Error: cannot load such file -- irb/easter-egg`

### ISSUE THREAD

**Issue by studiomohawk** [Homebrew/homebrew-core#50277]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

- Upgrade outdated formulas as my routine
- Trying to `brew cask install gpg-suite` 

## What happened (include command output)

`brew install <formula>` and `brew cask install <formula>` yields `Error: cannot load such file -- irb/easter-egg`

<details>
  <summary>Command output</summary>
  <pre>
brew install wget
Error: cannot load such file -- irb/easter-egg
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/Users/studiomohawk/.gem/ruby/2.6.0/gems/irb-1.2.2/lib/irb.rb:24:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:65:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:65:in `require'
/usr/local/Homebrew/Library/Homebrew/debrew/irb.rb:3:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/debrew.rb:4:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:12:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:3:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/info.rb:4:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/extend/os/mac/missing_formula.rb:4:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/extend/os/missing_formula.rb:3:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:196:in `singleton class'
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:7:in `<module:MissingFormula>'
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:6:in `<module:Homebrew>'
/usr/local/Homebrew/Library/Homebrew/missing_formula.rb:5:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:3:in `<top (required)>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/usr/local/Homebrew/Library/Homebrew/utils.rb:82:in `require?'
/usr/local/Homebrew/Library/Homebrew/commands.rb:28:in `valid_internal_cmd?'
/usr/local/Homebrew/Library/Homebrew/brew.rb:69:in `<main>'
  </pre>
</details>

<details>
  <summary>brew config</summary>
  <pre>
HOMEBREW_VERSION: 2.2.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4493f27db28be7f083737424cecd8edb863ec3f6
Last commit: 12 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 62043441c41c7311e960b8afa2c15ccc71cea9b5
Core tap last commit: 63 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_AWS_ACCESS_KEY_ID: set
HOMEBREW_AWS_SECRET_ACCESS_KEY: set
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
Clang: 11.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.3-x86_64
CLT: 11.0.33.17
Xcode: 11.3.1
XQuartz: 2.7.11 => /opt/X11
  </pre>
</details>

## What you expected to happen

- Able to install and or upgrade <formula> without error.

## Step-by-step reproduction instructions (by running `brew install` commands)

- I've uninstall brew itself and re-installed twice
- Run `brew install <formula>`

**Comment by eliezio** [comment]
I was having exactly the same issue. I found out that this issue is caused by a broken reference in the newest irb 1.2.2 installed under our `$HOME/.gem/ruby/2.6.0/gems` directory.
I could fix the error by reverting to v1.2.1 with:
```
gem uninstall --user-install irb -v1.2.2
gem install --user-install irb -v1.2.1
```

**Comment by studiomohawk** [comment]
Thanks for the info.

I've run same commands (with `sudo` on my side) and brew looks like working correctly.

**Comment by eliezio** [comment]
BTW the irb bug was fixed in v1.2.3 (see https://github.com/ruby/irb/commit/73cda56d2534694ef88b72e93ac32ffb5cd7f8be)
